clippy_lints/src/misc.rs

```diff
@@ -519,7 +519,8 @@ fn is_signum(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
 }
 
 fn is_float(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
-    matches!(walk_ptrs_ty(cx.tables.expr_ty(expr)).sty, ty::Float(_))
+    let value = walk_ptrs_ty(cx.tables.expr_ty(expr)).sty;
+    matches!(value, ty::Float(_)) || matches!(value, ty::Array(ty::Float, _))
```

Comment:
Hi, @flip1995. Opened a PR; it's not much but first step right. 😃 

I'm getting an issue trying to compare the value to: `ty::Array(ty::Float, _))`

The error is about mismatched types: _Ty::Float not a unit struct/variant or constant_. 

I've searched the code to find other examples of how to do it e.g where `ty::Slice` is used but I haven't had any luck.

Reply:
Thanks for working on this issue! 

This should probably fix it:
```suggestion
    matches!(value, ty::Float(_)) || matches!(value, ty::Array(ty::Float(_), _))
```

`ty::Float` by itself is just a "tuple variant", which is something like a function, which returns a `ty::TyKind {ty::TyKind::Float}`. As you may know, you can't match against functions, only against `unit struct/variant or constant` (aka `42`, `ty::Float(_)`, `MyStruct { field, .. }`, ...). That's why this _weird_ error.

See this example for clarification: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=eeb6ef489427bcdc1266eae46697d0e3)
```rust
#[derive(Debug)]
enum Foo {
    A(u8),
}

fn main() {
    let foo: fn(u8) -> Foo = Foo::A;
    let x = foo(10);
    
    println!("{:?}", x);
}
```

Reply:
Heya, I made the change and updated it to: `matches!(value, ty::Float(_)) || matches!(value, ty::Array(ty::Float(_), _))`

Unfortunately, I'm getting the same error.

When I replaced `ty::Float(_)` above with `_ty`, the error disappears but a couple of the tests fail. I did this to find a situation when the error isn't triggered.

Reply:
Hm ok, that's probably because of the macro, I don't really know how it works. How about 
```rust
matches!(value, ty::Array(ty, _)) && matches!(ty, ty::Float(_))
```

If that doesn't work either, I'll take a closer look at the macro 😉

Reply:
It should work though:
https://github.com/rust-lang/rust-clippy/blob/c55d38ed7ace06e4a3a5939b3c7c01045017bca4/clippy_lints/src/utils/mod.rs#L754-L757
🤔

Reply:
Thanks. I will take a look and get back. 😃

Reply:
Oh, looking at the docs again, the thing with only one macro won't work, since the structure of the enum variant `ty::Array` is `Array(ty::Ty, _)` and not `Array(ty::TyKind, _)`, so this: https://github.com/rust-lang/rust-clippy/pull/4343#discussion_r312456899 should work, when you exchange the second `ty` with `ty.node`.

Reply:
No worries at all. I'll make the change. :)